src/_guides/language/effective-dart/design_migrated.md

```diff
@@ -1315,6 +1463,26 @@ dynamic mergeJson(dynamic original, dynamic changes) => ...
 mergeJson(original, changes) => ...
 {% endprettify %}
 
+Note that it's OK to omit the type when Dart *successfully* infers `dynamic`.
+
+{:.good}
+<?code-excerpt "design_good.dart (infer-dynamic)"?>
+{% prettify dart tag=pre+code %}
+Map<String, dynamic> readJson() => ...
+
+void printUsers() {
+  var json = readJson();
+  var users = json['users'];
+  print(users);
+}
+{% endprettify %}
+
+Here, Dart infers `Map<String, dynamic>` for `json` and then from that infers
+`dynamic` for `users`. It's fine to leave `users` without a type annotation. The
+distinction is a little subtle. It's OK to allow inference to *propagate*
+`dynamic` through your code from a `dynamic` type annotation somewhere else, but
+you don't want it to summon `dynamic` *ex nihilo* when nothing in your code ever
```

Comment:
I'd never seen *ex nihilo* before, and I've seen a lot of Latin! I'm going to pretend I didn't see it, since it's probably against style but is totally cool (and doesn't really affect the meaning of the sentence).